src/types.ts

```diff
@@ -14,7 +14,9 @@ export type EventFrom<T extends (...args: any) => BusEvent> = ReturnType<T>;
 
 export type DispatchFn<E> = (a: E) => void;
 
-type UnsubscribeFn = () => any;
+export type UnsubscribeFn = () => any;
+
+export type SubscriptionDef<E extends BusEvent = BusEvent> = E["type"] | EventTypeDescriptor<E> | PredicateFn | string[]
```

Comment:
What about when you want to listen for multiple type definitions that are EventTypeDescriptors? Or you want to mix and match with predicates?

```ts
b.subscribe([myEvent, a => a.number > Math.PI, "*.foo"]);
```

Reply:
The idea of this type is to consolidate all possible ways of subscribing to events. As such, the functionality is the same it's just that we support string[] in addition to the existing ways of subscribing to events. Having read your comments. I think you have slightly something else in mind compared to #37. And I think looking from the perspective of the ts-bus, it makes absolutely sense. So yes, passing in an array with different subscription types provides the maximum flexibility. Basically, the type definition should probably be like.

```ts
export type SubscriptionDef<E extends BusEvent = BusEvent> = E["type"] | EventTypeDescriptor<E> | PredicateFn | string
```

And the bus subscribe method should accept an array.

```ts
subscribe = <T extends BusEvent>(
    subscription: SubscriptionDef<T>[],
    handler: (e: T) => void
  )
```

I will work a bit on the next iteration of that feature and provide the alpha version on which we can philosophize further :)